### DISCUSSION TITLE
Use graphql as peer dependency

### DISCUSSION THREAD

**Discussion by cbratschi** [payloadcms/payload#7633] [Q&A]
### Describe the Bug

Currently graphql is used as a normal dependency which causes version errors if a slightly newer version is installed. The package should always be used as a peer dependency:

https://github.com/graphql/graphql-js/issues/594#issuecomment-1189850052

### To Reproduce

Install latest graphql version and graphql complains about having different version being installed.

### Payload Version

2.26.0

### Adapters and Plugins

db-mongodb, bundler-webpack, live-preview

**Comment by jmikrut** [comment]
Hey @cbratschi — you're correct with this issue. We've already moved to the pattern that is suggested in 3.x, but in 2.x, that would be a breaking change. For this reason, we are unlikely to back-port it because it would require a major release.

Is there a reason that you need to have GraphQL installed locally? You should use the Payload-provided GraphQL instance if you are building custom GraphQL queries / mutations, which will solve for the need to have it locally installed.

Would love to learn more but will convert this to a discussion and we can continue the conversation there!

**Reply by cbratschi** [reply]
The main reason is that it is being used in a monorepo where other projects need the graphql package.
For now we can use the same version Payload uses until Payload 3.x is available.
